### PR TITLE
Sd 109 user wallet stats

### PR DESCRIPTION
**Pull request by githinho**

### CODE REVIEW

File: src/main/kotlin/com/ampnet/walletservice/grpc/blockchain/BlockchainService.kt [hunk 1]
```diff
@@ -31,4 +32,5 @@ interface BlockchainService {
     fun generateTransferPlatformManager(address: String, coop: String): TransactionData
     fun getSellOffers(coop: String): List<SellOfferData>
     fun deployCoopContract(coop: String, address: String)
+    fun getUserWalletsWithInvestment(coop: String): List<UserWalletsForCoopAndTxTypeResponse.WalletWithHash>
```
Comment (domagojlatecki) [inline src/main/kotlin/com/ampnet/walletservice/grpc/blockchain/BlockchainService.kt:35]: Import sub-class `WalletWithHash` or create a type alias that is shorter than `UserWalletsForCoopAndTxTypeResponse.WalletWithHash`

File: src/main/kotlin/com/ampnet/walletservice/grpc/blockchain/BlockchainServiceImpl.kt [hunk 2]
```diff
@@ -461,6 +464,26 @@ class BlockchainServiceImpl(
         }
     }
 
+    @Throws(GrpcException::class, GrpcHandledException::class)
+    override fun getUserWalletsWithInvestment(coop: String): List<UserWalletsForCoopAndTxTypeResponse.WalletWithHash> {
```
Comment (domagojlatecki) [inline src/main/kotlin/com/ampnet/walletservice/grpc/blockchain/BlockchainServiceImpl.kt:468]: Import sub-class or use type alias

File: src/test/kotlin/com/ampnet/walletservice/grpc/GrpcWalletServerTest.kt [hunk 3]
```diff
@@ -80,6 +90,33 @@ class GrpcWalletServerTest : TestBase() {
         }
     }
 
+    @Test
+    fun mustReturnOwnersOfApprovedDeposits() {
+        suppose("Wallets exist") {
+            testContext.uuids = listOf(UUID.randomUUID(), UUID.randomUUID())
+            testContext.deposits = testContext.uuids.map { createDeposit(it) }
+            Mockito.`when`(
```
Comment (domagojlatecki) [inline src/test/kotlin/com/ampnet/walletservice/grpc/GrpcWalletServerTest.kt:98]: Use BDD Mockito's `given`

File: src/test/kotlin/com/ampnet/walletservice/grpc/GrpcWalletServerTest.kt [hunk 4]
```diff
@@ -80,6 +90,33 @@ class GrpcWalletServerTest : TestBase() {
...
+                .setCoop(COOP)
+                .build()
+
+            @Suppress("UNCHECKED_CAST")
+            val streamObserver = Mockito.mock(StreamObserver::class.java) as StreamObserver<OwnersResponse>
+            grpcServer.getOwnersWithDeposit(request, streamObserver)
+            val response = OwnersResponse.newBuilder()
+                .addAllOwnersUuids(testContext.uuids.map { it.toString() })
+                .build()
+            Mockito.verify(streamObserver).onNext(response)
```
Comment (domagojlatecki) [inline src/test/kotlin/com/ampnet/walletservice/grpc/GrpcWalletServerTest.kt:114]: Use BDD Mockit's `then`

**Review by mightymatth — APPROVED** [review]

**Review by domagojlatecki — APPROVED** [review]

**Review by windhustler — APPROVED** [review]

**Pull request merged**